packages/multiple-did-identities/src/__tests__/did-methodfactory.spec.ts

```diff
@@ -19,15 +26,130 @@ describe('DidMethodFactory', () => {
   });
 
   describe('generateDid', () => {
-    it('should generate a DID using DidKeyMethod', async () => {
+    it('should generate a DID using DidKeyMethod when method is key', async () => {
       const result = await DidMethodFactory.generateDid(DIDMethodName.Key);
 
-      // Check that the generated DID is in the expected format
-      expect(result.did).toMatch(/^did:key:z[1-9A-HJ-NP-Za-km-z]+$/);
-      expect(result.publicKey.kty).toBe('OKP');
-      expect(result.publicKey.crv).toBe('Ed25519');
-      expect(result.publicKey.x).toBeDefined();
-      expect(result.privateKey.d).toBeDefined();
+      // Type guard to check if result is DIDKeyPair
+      if ('publicKey' in result && 'privateKey' in result) {
+        expect(result.publicKey.kty).toBe('OKP');
+        expect(result.publicKey.crv).toBe('Ed25519');
+        expect(result.publicKey.x).toBeDefined();
+        expect(result.privateKey.d).toBeDefined();
+      } else {
+        throw new Error(
+          "Result is not of type DIDKeyPair with 'publicKey' and 'privateKey'",
+        );
+      }
+    });
+
+    it('should throw an error if methodType is provided with DIDMethodName.Key', async () => {
+      await expect(
+        DidMethodFactory.generateDid(
+          DIDMethodName.Key,
+          'method1' as PeerGenerationMethod,
+        ),
+      ).rejects.toThrow(
+        'methodType should not be specified for DIDMethodName.Key',
+      );
+    });
+
+    it('should throw an error if methodType is not provided with DIDMethodName.Peer', async () => {
+      await expect(
+        DidMethodFactory.generateDid(DIDMethodName.Peer),
+      ).rejects.toThrow('methodType must be specified for DIDMethodName.Peer');
+    });
+
+    it('should create an instance of DidPeerMethod when DIDMethodName.Peer is provided', () => {
+      const method = DidMethodFactory.create(DIDMethodName.Peer);
+      expect(method).toBeInstanceOf(DidPeerMethod);
+    });
+
+    it('should create an instance of DidKeyMethod when DIDMethodName.Key is provided', () => {
+      const method = DidMethodFactory.create(DIDMethodName.Key);
+      expect(method).toBeInstanceOf(DidKeyMethod);
+    });
+
+    it('should throw an error for an unsupported DID method', () => {
+      expect(() =>
+        DidMethodFactory.create('unsupported' as DIDMethodName),
+      ).toThrow('Unsupported DID method: unsupported');
+    });
+
+    const peerMethods: PeerGenerationMethod[] = [
+      'method0',
+      'method1',
+      'method2',
```

Comment:
I suggest we use enums for peer method types. Will avoid string typing issues. 

export enum PeerMethodTypes {
   Method0 = 'method0',
 }